src/Acuminator/Acuminator.Analyzers/Resources.resx

```diff
@@ -799,4 +799,16 @@ Reason: {2}</value>
   <data name="SuppressDiagnosticWithCommentNonNestedCodeActionTitle" xml:space="preserve">
     <value>Suppress the {0} diagnostic with Acuminator in a comment</value>
   </data>
-</root>
+  <data name="PX1078MessageFormat_TypesOfDacFieldAndReferencedFieldMismatch" xml:space="preserve">
+    <value>DAC field {0} and field referenced by PXSelectorAttribute {1} have different types. Expected type declaration for the DAC field is {2}.</value>
```

Comment:
@acasj could you please provide filled example of how the diagnostic message should look with all format arguments filled.

Reply:
Yes, please. It will be useful for the doc team.

Reply:
`DAC field PaymentTermsId and referenced field substitutionID have different length specified in PXDBString attribute. Expected length for DAC field is 25.` for mismatched length
`DAC field PaymentTermsId and referenced field substitutionID have different types. Expected type declaration for the DAC field is [PXDBString(25, InputMask = "", IsKey = false)].` for mismatched data type

Reply:
Thanks! 
I have a few remarks regarding the messages:

1. The usage of attribute names in the first message:
```
DAC field PaymentTermsId and referenced field substitutionID have different length specified in PXDBString attribute. Expected length for DAC field is 25. 
```
I'm not sure, if we need to specify the name of attribute here. Imho, it's ok to have it like this:
```
DAC field PaymentTermsId and referenced field substitutionID have different length specified in the data type attribute. Expected length for DAC field is 25. 
```
This will cover cases when local and external DAC fields have different data type attributes which is possible. 

2. I think it is wrong to use data type attribute in the second string:
```
DAC field PaymentTermsId and referenced field substitutionID have different types. Expected type declaration for the DAC field is [PXDBString(25, InputMask = "", IsKey = false)]
```
Here are my reasons:
- Mentioning the normal CLR / C# type and a need to change the property type will be easier to understand for many developers
- There is not enough context for Acuminator to provide a recommendation about attribute declaration for developers. Acuminator can report incorrect type, can also tell the correct length for string DAC fields, but that's all. We don't know which data attribute should be placed on the field because it is up to the application developer. It may be `PXDBString`, or may be something derived from it, or perhaps some aggregator attribute.
- Data type attributes do not define type of some data themselves. So, strictly speaking a declaration of a data type attribute is not a type declaration. 

3. The name of DAC fields should have the same casing. I believe, that the PascalCase would be better here because this diagnostic checks the declaration of DAC properties, not DAC BQL fields.

4. Perhaps it would be useful to enclose field names and length value in quotes, but I'm not sure about this. There are other similar strings from other diagnostics that don't do this. I will leave it up to you and the doc team